sigstore/_cli.py

```diff
@@ -149,12 +165,25 @@ def _add_shared_verify_input_options(group: argparse._ArgumentGroup) -> None:
         default=os.getenv("SIGSTORE_BUNDLE"),
         help=("The Sigstore bundle to verify with; not used with multiple inputs"),
     )
+
+    def file_or_digest(arg: str) -> Hashed | Path:
+        if arg.startswith("sha256:"):
+            digest = bytes.fromhex(arg[len("sha256:") :])
+            if len(digest) != 32:
+                raise ValueError()
```

Comment:
Where will this be caught? We might want to provide a more informative error message here.

Reply:
`argparse` [catches](https://docs.python.org/3/library/argparse.html#type) it:
> The argument to type can be any callable that accepts a single string. If the function raises [ArgumentTypeError](https://docs.python.org/3/library/argparse.html#argparse.ArgumentTypeError), [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError), or [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError), the exception is caught and a nicely formatted error message is displayed. No other exception types are handled.

The only problem is that it ignores the error message, so whatever message I put in `ValueError("mymsg")`, the shown error is always:
```
sigstore verify identity: error: argument FILE_OR_DIGEST: invalid file_or_digest value: 'sha256:3072b971b74902ac37ee452649efbaaeca6cd80c93cef5e27c1f491d7aa2ab443'
```

Reply:
Hmm, would be nice to have a more specific message but this should be fine!